lib/syskit/task_context.rb

```diff
@@ -419,6 +442,25 @@ def commit_properties_if_needed(*args)
             end
         end
 
+        def start!
+            return super unless read_only?
+
+            event(:start).emit
+        end
+
+        def stop!
+            return super unless read_only?
+
+            event(:stop).emit
+        end
+
+        def check_emission_validity(event) # :nodoc:
+            return super unless read_only?
+
+            super unless running? && event.symbol == :start
+            super if started? && event.symbol == :stop
+        end
+
```

Comment:
And no. Your job is to make sure the read only task behaves properly. Not to change the definition of "behaving properly"